src/clients/asset-client.js

```diff
@@ -14,42 +18,93 @@ class AssetClient {
     this.directory = directory
   }
 
+  /**
+   *
+   * @param {File} file A file object.
+   * @param {Object} options An options object allowing to specify onProgress and doAbort functions.
+   * onProgress handles upload progress in app utilizing this client. doAbort function returning true/false indicating the need to abort upload.
```

Comment:
I don't like the word `abort`. Can we rename it to cancel please. 🙏 

`doAbort` => `doCancel`

Reply:
Unless you feel completely the opposite, then let's keep `doAbort`. Both names are correct.